_posts/2019-04-17-accessing-kafka-part-1.md

```diff
@@ -0,0 +1,108 @@
+---
+layout: post
+title:  "Accessing Kafka: Part 1 - Introduction"
+date: 2019-04-17
+author: jakub_scholz
+---
+
+Scalability is one of the flagship features of Apache Kafka.
+It is achieved by partitioning the data and distributing them across multiple brokers.
+Such data sharding has also a big impact on how Kafka clients connect to the brokers.
+This is especially visible when Kafka is running within a platform like Kubernetes but is accessed from outside of that platform.
+This blog-series will explain how Kafka and its clients work and how Strimzi deals with it to make it accessible for clients running outside of Kubernetes.
+
+<!--more-->
+
+It would, of course, be insufficient to just shard the data into partitions.
+The ingress and egress data traffic also needs to be properly handled.
+The clients writing to or reading from a given partition have to connect directly to the leader broker which is hosting it.
+Thanks to the clients connecting directly to the individual brokers, the brokers don't need to do any forwarding of data between the clients and other brokers.
+That helps to significantly reduce the amount of work the brokers have to do and the amount of traffic flowing around within the cluster.
+The only data traffic between the different brokers is due to replication, when the follower brokers are fetching data from the lead broker for a given partition.
+That makes the data shards independent on each other and that makes Kafka scale so well.
+
+![Clients connecting to partitions]({{ "/assets/2019-04-17-connecting-to-leader.png" }})
+
+But how do the clients know where to connect?
+
+# Kafka's discovery protocol
+
+Kafka has its own discovery protocol.
+When a Kafka client is connecting to the Kafka cluster, it first connects to any broker which is member of the cluster and asks it for _metadata_ for one or more topics.
+These _metadata_ contain the information about the topics, its partitions and brokers which are responsible for them.
```

Comment:
Do we want to enforce here what you already said for example adding "...responsible for them (the leader partitions)"?